src/med_bench/estimation/base.py

```diff
@@ -0,0 +1,496 @@
+
+        # Initialisation
+        (
+            mu_1mx,  # E[Y|T=1,M,X]
+            mu_0mx,  # E[Y|T=0,M,X]
+        ) = [np.zeros(n) for _ in range(2)]
+
+        t0, m0 = np.zeros((n, 1)), np.zeros((n, 1))
+        t1, m1 = np.ones((n, 1)), np.ones((n, 1))
+
+        x_t_m = np.hstack([x, t.reshape(-1, 1), m])
+        x_t1_m = np.hstack([x, t1.reshape(-1, 1), m])
+        x_t0_m = np.hstack([x, t0.reshape(-1, 1), m])
+
+        test_index = np.arange(n)
+        ind_t0 = t[test_index] == 0
+
+        reg_param_grid = {}
+
+        # estimate conditional mean outcomes
+        regressor_y = GridSearchCV(self.regressor, reg_param_grid)
+
+        self.regressors = {}
+
+        # mu_tm model fitting
+        self.regressors["y_t_mx"] = clone(regressor_y).fit(x_t_m, y)
+
+        # predict E[Y|T=t,M,X]
+        mu_1mx[test_index] = self.regressors["y_t_mx"].predict(
+            x_t1_m[test_index, :])
+        mu_0mx[test_index] = self.regressors["y_t_mx"].predict(
+            x_t0_m[test_index, :])
+
+        for i, b in enumerate(np.unique(m)):
+            mb = m1 * b
+
+            mu_1bx, mu_0bx = [np.zeros(n) for h in range(2)]
+
+            # predict E[Y|T=t,M=m,X]
+
+            x_t1_mb = np.hstack([x, t1.reshape(-1, 1), mb])
+            x_t0_mb = np.hstack([x, t0.reshape(-1, 1), mb])
+
+            mu_0bx[test_index] = self.regressors["y_t_mx"].predict(
+                x_t0_mb[test_index, :]
+            )
+            mu_1bx[test_index] = self.regressors["y_t_mx"].predict(
+                x_t1_mb[test_index, :]
+            )
+
+            # E[E[Y|T=1,M=m,X]|T=t,X] model fitting
+            self.regressors["reg_y_t1m{}_t0".format(i)] = clone(regressor_y).fit(
+                x[test_index, :][ind_t0, :], mu_1bx[test_index][ind_t0]
+            )
+            self.regressors["reg_y_t1m{}_t1".format(i)] = clone(regressor_y).fit(
+                x[test_index, :][~ind_t0, :], mu_1bx[test_index][~ind_t0]
+            )
+
+            # E[E[Y|T=0,M=m,X]|T=t,X] model fitting
+            self.regressors["reg_y_t0m{}_t0".format(i)] = clone(regressor_y).fit(
+                x[test_index, :][ind_t0, :], mu_0bx[test_index][ind_t0]
+            )
+            self.regressors["reg_y_t0m{}_t1".format(i)] = clone(regressor_y).fit(
+                x[test_index, :][~ind_t0, :], mu_0bx[test_index][~ind_t0]
+            )
+
+        return self
+
+    def _estimate_mediator_probability(self, t, m, x, y):
+        """
+        Estimate mediator density f(M|T,X)
+
+        Returns
+        -------
+        f_m0, array-like, shape (n_samples)
+            probabilities f(M|T=0,X)
+        f_m1, array-like, shape (n_samples)
+            probabilities f(M|T=1,X)
+        """
+        n = len(y)
+
+        t0 = np.zeros((n, 1))
+        t1 = np.ones((n, 1))
+
+        m = m.ravel()
+
+        t0_x = np.hstack([t0.reshape(-1, 1), x])
+        t1_x = np.hstack([t1.reshape(-1, 1), x])
+
+        fm_0 = self._classifier_m.predict_proba(t0_x)[:, 1]
```

Comment:
it seems false, for me it predicts P(M=1|T=0, X=x) and not P(M=m|T=0, X=x), should be
```
fm_0 = self._classifier_m.predict_proba(t0_x)[:, m]
```

no? and same at the line after